app/src/main/java/com/venuenext/venuenextsdkdemo/settings/VenueNextFlowActivity.kt

```diff
@@ -46,8 +50,14 @@ class VenueNextFlowActivity : AppCompatActivity(), TicketingInterface, WalletInt
             if (navId != 0) {
                 navController.popBackStack()
 
+                val navBundle = inboundBundle ?: bundleOf("showBackButton" to true)
                 // Show the back button in these flows
-                navController.navigate(navId, bundleOf("showBackButton" to true))
+                navController.navigate(navId, navBundle)
+            } else {
+                // When back-navigating, this handles the scenario where the hosting Activity is
+                // re-launched. In our scenario, we start the MainActivity again, which drops the
+                // user back onto the StandsFragment. Apply your own flow/business logic here.
+                startActivity(Intent(this, MainActivity::class.java))
```

Comment:
Can we just do finish? Or does that kill the entire app?

Reply:
I tested this first - `finish` does kill the entire app, so I'm using ultimately what Adept will be using in their hosting Activity.